sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -968,6 +968,7 @@ object TypeCoercion extends TypeCoercionBase {
       case (StringType, target: NumericType) => target
       case (StringType, DateType) => DateType
       case (StringType, TimestampType) => TimestampType
+      case (StringType, AnyTimestampType) => TimestampType
```

Comment:
We need this for the default implicit cast from string to timestamp type. This regression is found on running extract.sql.